packages/mimir/src/rules/no-useless-spread.ts

```diff
@@ -0,0 +1,24 @@
+import * as ts from 'typescript';
+import { AbstractRule, Replacement } from '@fimbul/ymir';
+
+const FAILURE_STRING = 'Using the spread operator here is not necessary.';
+
+export class Rule extends AbstractRule {
+    public apply() {
+        for (const node of this.context.getFlatAst())
+            if (
+                (node.kind === ts.SyntaxKind.SpreadElement &&
+                    (<ts.SpreadElement>node).expression.kind === ts.SyntaxKind.ArrayLiteralExpression) ||
+                (node.kind === ts.SyntaxKind.SpreadAssignment &&
+                    (<ts.SpreadAssignment>node).expression.kind === ts.SyntaxKind.ObjectLiteralExpression)
+            )
+                this.addFailureAtNode(node, FAILURE_STRING, removeUselessSpread(<ts.SpreadElement>node));
+    }
+}
+
+function removeUselessSpread(node: ts.SpreadElement): Replacement[] {
```

Comment:
the parameter should be `ts.SpreadElement | ts.SpreadAssignment` to prevent bugs in the future